src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,113 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import duke.utils.DukeException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.List;
+import java.util.Scanner;
+
+public class Storage {
+    public final static DukeException ERROR_DB = new DukeException("Error loading database.");
+    private final RandomAccessFile raf;
+    private final File txt;
+
+    public Storage(TaskList db) throws DukeException {
+        try {
+            txt = new File("duke.txt");
+            raf = new RandomAccessFile(txt, "rwd");
+            while (raf.getFilePointer() < raf.length()) {
+                Task t = parse(raf.readLine());
+                if (t != null) {
+                    db.add(t);
+                }
+            }
+            update(db);
+        } catch (IOException e) {
+            throw ERROR_DB;
+        }
+    }
+
+    public Task parse(String line) throws DukeException {
+        try {
+            String[] args = line.split("( \\| )", 3);
+            if (args.length < 3) {
+                return null;
+            }
+            boolean done = Integer.parseInt(args[1]) == 1;
+            switch (args[0]) {
+            case "T":
```

Comment:
Rather than just the char here, perhaps you could add a char constant to the different Task classes to enhance readability? Also to make it easier to edit in the future when task types with overlapping char representations are created! :~)

Reply:
Thank you for the suggestion, will implement